rust/src/lib.rs

```diff
@@ -217,19 +217,31 @@ Accept-Encoding: gzip,deflate\r
     pub fn lex(body: &[u8]) -> String {
         // 13. Print content
         let mut in_angle = false;
-        let mut ret = String::new();
+        let mut in_body_tag = false;
+        let mut out = String::new();
+        let mut tag = String::new();
```

Comment:
because `out` and `body` are alive until the function end, they don't need to be `String` but `&[u8]`.

Reply:
I'll do this on another branch.

Reply:
Please follow code review if possible in a single PR.